File: Data Structures & Algorithms/combination-target-sum/submission-2.py
class Solution:
    def combinationSum(self, nums: List[int], target: int) -> List[List[int]]:
        res=[]
        nums.sort()
        curr=[]
        def backtrack(index,target,curr,nums):
            if target==0:
                res.append(curr[:])
                return
            for i in range(index,len(nums)):
                if target-nums[i]<0:
                    break
                curr.append(nums[i])
                backtrack(i,target-nums[i],curr,nums)
                curr.pop()
        backtrack(0,target,curr,nums)
        return res

File: Data Structures & Algorithms/find-duplicate-integer/submission-2.java
class Solution {
    public int findDuplicate(int[] nums) {
        int s=0,f=0;
        while(true){
            s=nums[s];
            f=nums[nums[f]];
            if(s==f) break;
        }
        s=0;
        while(s!=f){
            s=nums[s];
            f=nums[f];
        }
        return s;
    }
}
